concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limit/window/AverageSampleWindowFactory.java

```diff
@@ -0,0 +1,36 @@
+package com.netflix.concurrency.limits.limit.window;
+
+public class AverageSampleWindowFactory implements SampleWindowFactory<ImmutableAverageSampleWindow> {
+    private AverageSampleWindowFactory() {}
+
+    public static AverageSampleWindowFactory create() {
+        return new AverageSampleWindowFactory();
+    }
+
+    @Override
+    public ImmutableAverageSampleWindow newInstance() {
+        return new ImmutableAverageSampleWindow();
+    }
+
+    @Override
```

Comment:
Let's keep these methods as part of the SampleWindow abstraction and just have newInstance() in the factory.

Reply:
I thought it would be better to keep all the "creation logic" on the side of factories, so sample windows know nothing about it, but wasn't completely sure about it